Word-to-PDF-Conversion/Get-missing-fonts-for-pdf-conversion/.NET/Get-missing-fonts-for-pdf-conversion/Program.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Syncfusion.DocIO;
+using Syncfusion.DocIO.DLS;
+using Syncfusion.DocIORenderer;
+using Syncfusion.Pdf;
+
+namespace Get_missing_fonts_for_pdf_conversion
+{
+    class Program
+    {
+        // List to store names of fonts that are not installed
+        static List<string> fonts = new List<string>();
+        static void Main()
+        {
+            // Open the existing DOCX file stream
+            using (FileStream docStream = new FileStream(Path.GetFullPath(@"../../../Data/Input.docx"), FileMode.Open, FileAccess.Read))
+            {
+                // Load the file stream into a Word document
+                using (WordDocument wordDocument = new WordDocument(docStream, FormatType.Docx))
+                {
+                    // Hook the font substitution event to detect missing fonts
+                    wordDocument.FontSettings.SubstituteFont += FontSettings_SubstituteFont;
+
+                    // Instantiate DocIORenderer for Word to PDF conversion
+                    using (DocIORenderer render = new DocIORenderer())
+                    {
+                        // Convert Word document into PDF document
+                        PdfDocument pdfDocument = render.ConvertToPDF(wordDocument);
+
+                        // Save the PDF document to output stream
+                        using (FileStream outputStream = new FileStream(Path.GetFullPath(@"../../../Data/Result.pdf"), FileMode.OpenOrCreate, FileAccess.ReadWrite))
+                        {
+                            pdfDocument.Save(outputStream);
+                        }
+                    }
+                }
+            }
+
+            // Check and output the fonts that are not installed
+            if (fonts.Count > 0)
+            {
+                Console.WriteLine("Fonts not installed in environment:");
+                foreach (string font in fonts)
+                    Console.WriteLine(font);
+            }
+            else
+            {
+                Console.WriteLine("Document used fonts are all installed.");
```

Comment:
"Fonts used in Word document are available in environment"

Reply:
Modified